api-client/src/utils/getNavigationDeepLinks.ts

```diff
@@ -0,0 +1,29 @@
+import LocationModel from '../models/LocationModel'
+
+const getNavigationDeepLinks = (
+  location: LocationModel,
+  platform: 'web' | 'android' | 'ios' | string
+): string | null => {
+  if (!location.location || !location.coordinates) {
+    return null
+  }
+  const long = location.coordinates[0]
+  const lat = location.coordinates[1]
+  const android = `geo:${lat},${long}?q=${location.location}`
+
+  switch (platform) {
+    case 'web': {
+      const baseUrl = `https://maps.google.com?q=`
+      const isAndroid = /Android/i.test(navigator.userAgent)
+      return isAndroid ? android : `${baseUrl}${location.location},${lat},${long}`
```

Comment:
Wouldn't it be way easier if his check happened in the place where this code is called in web and then just pass `android` or `web` depending on the result?
I also think that would be the way I'd expect to use it. If I pass `web` as platform I'd not expect that there is internally a check for the user agent and I may get an android deep link instead of an url.

Reply:
Fine for me, will change.

Reply:
As in my first comment I opened discussion for this solution: https://github.com/digitalfabrik/integreat-app/pull/694#issuecomment-1162834448